### PR TITLE
coc-pyright: migrate from nodePackages

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Migrates this package from the nodePackages set, and updates the vimPlugins set so that it pulls from the top-level package instead of the one in the nodePackages set.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -342,6 +343,12 @@
     src = "${coc-diagnostic}/lib/node_modules/coc-diagnostic";
   };
 
+  coc-pyright = buildVimPlugin {
+    pname = "coc-nginx";
```
Comment (smancill) [inline pkgs/applications/editors/vim/plugins/overrides.nix:347]: Shouldn't it be `coc-pyright`?

**Comment by PerchunPak** [comment]
Why though?

File: pkgs/applications/editors/vim/plugins/overrides.nix (hunk 1 above)
Comment (pyrox0) [reply pkgs/applications/editors/vim/plugins/overrides.nix]: Yes, thank you for noticing that!

**Comment by PerchunPak** [comment]
Still, why move this package from `nodePackages`?

**Comment by pyrox0** [comment]
> Still, why move this package from `nodePackages`?

See #229475

**Comment by smancill** [comment]
I think this is ok to merge, but I'd like another person agreeing with it. I am not much aware with the situation about node packages.

**Comment by PerchunPak** [comment]
Result of `nixpkgs-review pr 327171` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>coc-pyright</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

File: pkgs/by-name/co/coc-pyright/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+}:
+
+buildNpmPackage {
+  pname = "coc-pyright";
+  version = "1.1.371";
```
Comment (PerchunPak) [inline pkgs/by-name/co/coc-pyright/package.nix:9]: Looks like there is a new release, how about updating?
https://github.com/fannheyward/coc-pyright/commit/7a765863fab5b3345018377b47203e8cfb30b4e3

**Review by PerchunPak — APPROVED** [review]
LGMT, though it would be nice if someone could test this on their coc.nvim installation

**Review by natsukium — APPROVED** [review]
LGTM. Let's update it in another PR.

**Pull request merged**